js/base.js

```diff
@@ -51,6 +51,13 @@ export class API {
         const topic = options.topic || name.split(".", 1)[0];
 
         await this.producer.produce(topic, event, options);
+
+        const globalDiffusion =
+            options.globalDiffusion === undefined
+                ? this.producer.globalDiffusion
+                : options.globalDiffusion;
+        const globalTopic = this._globalTopic(topic);
+        if (globalDiffusion && globalTopic) await this.trigger(globalTopic, event, options);

```

Comment:
Ok, in a closer look, trigger takes 

```js
async trigger(name, payload, options = {})
```

So when you call

```js
await this.trigger(globalTopic, event, options)
```

You are saying the name is `globalTopic` which is fine because this line still works:

```js
const topic = options.topic || name.split(".", 1)[0];
```

But you are saying `payload = event` which is not tru because now you have

```js
const event = {
            name: name,
            origin: options.origin || null,
            hostname: options.hostname || os.hostname(),
            datatype: options.datatype || "json",
            timestamp: options.timestamp || Date.now(),
            payload: payload
        };
```

Where payload is the previous event, right?

Reply:
True, good catch